### PR TITLE
update .npmrc configuration in setup_nexus_npm

### PR DESCRIPTION
**Pull request by tlotempus**
previously, we were only authorizing npm-txm, this also adds npm

### CODE REVIEW

**Review by beiriannydd — APPROVED** [review]
since we're in here, can we look at if inputs is empty and if it is use secrets instead?
If that's a step too far here, I'm happy to approve.

This PR brings the file content in line with the latest NPM changes.

File: setup-nexus-npm/action.yml [hunk 1]
```diff
@@ -11,7 +11,8 @@ runs:
   steps:
     - run: |
         cat << EOF > ~/.npmrc
-        registry=https://${{ inputs.endpoint }}/repository/npm-txm
-        _auth=$(echo -n ${{ inputs.login }}:${{ inputs.password }} | base64)
+        registry = https://${{ inputs.endpoint }}/repository/npm/
+        @txm:registry=https://${{ inputs.endpoint }}/repository/npm-txm/
+        //nexus.txm.tools/repository/:_auth=$(echo -n ${{ inputs.login }}:${{ inputs.password }} | base64)
```
Comment (raven-txm) [inline setup-nexus-npm/action.yml]: If ${{ inputs.endpoint }} is nexus.txm.tools, why don't we use it on this line as well?

**Review by raven-txm — CHANGES_REQUESTED** [review]

File: setup-nexus-npm/action.yml (hunk 1 above)
Comment (tlotempus) [reply setup-nexus-npm/action.yml]: sure, we can do that

**Comment by tlotempus** [comment]
@beiriannydd @raven-txm - given that this is "setup-nexus-npm", I'm not sure we need to make the endpoint an optional parameter.  What would you think of just removing the input option for endpoint and hard coding `nexus.txm.tools`?  I mean what other value would it ever be, particularly given the fact that nexus is proxying public npm repos?

**Comment by beiriannydd** [comment]
I think the option is nice, but defaulting to nexus.txm.tools is fine. Reasons for having an option:
Nexus is hard down with no ETA
Nexus migration simplification
Testing dev nexus

**Comment by tlotempus** [comment]
@beiriannydd - 👍  on the optional value with default value for `endpoint`.
Regarding username and password - the calling workflow would be passing in repo specific secrets as we intend to do with txm-load-testing, FETT, etc...
Given that username and password are currently `required: true`, would that not prevent empty values here?

**Comment by beiriannydd** [comment]
My thoughts on this as discussed elsewhere, it would be great to have short-term credentials rotated more often.  This will be easier to do if centralized.  If we specify a secret in the actions-hub repo and make username/password not required, it does mean anyone with access to the action can publish but I also think that's probably fine too since we have to approve repos?

**Comment by raven-txm** [comment]
> we have to approve repos?

Since when have we had to approve repos?

**Comment by raven-txm** [comment]
> What would you think of just removing the input option for endpoint and hard coding `nexus.txm.tools`? I mean what other value would it ever be, particularly given the fact that nexus is proxying public npm repos?

We already have reasons to keep this parameter around discussed. Defaulting it to nexus.txm.tools is fine. However, we use the parameter in two lines, and not in the third - for consistency we should use it in all three places.

The larger discussion probably belongs in a ticket, not in this PR?

**Comment by beiriannydd** [comment]
You've always had to grant access to actions-hub to the repo that wants to use it because it isn't the .github repo and therefore shared to the org.

**Comment by raven-txm** [comment]
Ah, that explains it - I haven't had to do this, so I thought you meant I couldn't create a repo at-will. Makes sense now. XD

**Comment by beiriannydd** [comment]
actually, thinking carefully about John's comments
the credentials and repo should all agree.
we should not use our shared secret if the repository is outside txm.tools.

Sorry for adding complexity to this, @tlotempus

**Comment by tlotempus** [comment]
> actually, thinking carefully about John's comments the credentials and repo should all agree. we should not use our shared secret if the repository is outside txm.tools.

@beiriannydd - please clarify?  My plan was to make username and password `required: false` and default their values to `NEXUS_USER_CI_NPM` and `NEXUS_PASSWORD_CI_NPM` which are defined as secrets on this repo.  Will that work or no?

My assumption (and certainly in the case of OE repos) is that if these values get defined in the calling workflow, they will use the same secrets...

**Comment by beiriannydd** [comment]
That seems like a good idea, however, if the passed endpoint != txm.tools we should only use a passed in credential and not send our internal credential to a 3rd party

**Comment by tlotempus** [comment]
ugh... this gets exponentially more complicated now because in the case where the endpoint is not our nexus, we can't make assumptions about the format of the third-party endpoint, which I think means we would need a fourth conditional input for the third-party npm endpoint.  It almost makes more sense to create a `setup-third-party-npm` action and separate the concerns...

**Comment by beiriannydd** [comment]
I support that @tlotempus

**Review by raven-txm — APPROVED** [review]

**Pull request merged**